backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/rest/service/SpecificationDataService.kt

```diff
@@ -74,24 +77,52 @@ class SpecificationDataService @Autowired constructor(
             null
         }
 
-    fun extractId(json: String): String? {
-        val parsedJson = objectMapper.readTree(json)
-        val jsonNode = parsedJson.at("/info/x-api-id")
-        return if (jsonNode.isMissingNode) {
+    fun extractId(json: String): String? =
+        try {
+            val parsedJson = objectMapper.readTree(json)
+            val jsonNode = parsedJson.at("/info/x-api-id")
+            if (jsonNode.isMissingNode) {
+                null
+            } else {
+                jsonNode.asText()
+            }
+        } catch (exception: IOException) {
             null
+        }
+
+    fun makeSpecificationMetaData(fileContent: String,
```

Comment:
If I understand that correctly, the decision whether it is `GRAPHQL` or `OPENAPI` hinges on whether there is `VersionFileMetaData`. To be honest, I find that a bit opaque and unexpected. Do you think there is a clearer way?

Reply:
I completely agree, and have been thinking about changing it. However maybe instead the language enum itself here is misleading. In my mind, the distinction ApiCenter cares about (when uploading) is whether or not the metadata can be sourced from the file contents, or needs to be additionally supplied by the user uploading the file --> ie. whether or not the frontend sends metadata. This could apply to other formats which do not include metadata required by ApiCenter in their specifications. The enum was initially added to simply choose between the SwaggerUI and GraphiQL. Perhaps adding a map from ApiLanguage values to whether they need metadata would make the distinction clear.